### ISSUE TITLE
DSM 6.2 support

### ISSUE THREAD

**Issue by ymartin59** [SynoCommunity/spkrepo#17]
INFO package property `firmware` is deprecated and replaced by `os_min_vers`.
It is expected Package Center - Repository API has evolved the same way, with effect no package is visible.
Strange behavior: installed package is upgradable
Optional extension: download count

**Comment by ymartin59** [comment]
From repository access log, it looks like `micro` parameter changed semantic:
```
172.68.94.127 - - [27/Jun/2018:21:39:26 +0200] "GET /?package_update_channel=beta&unique=synology_armada38x_ds216j&build=15284&language=enu&major=6&micro=7&arch=armada38x&minor=1&timezone=Amsterdam&nano=0 HTT
P/1.1" 200 83637 "-" "synology_armada38x_ds216j DSM6.1-15284 (package)"
162.158.202.103 - - [27/Jun/2018:21:39:26 +0200] "GET /?package_update_channel=stable&unique=synology_avoton_415%2B&build=23739&language=enu&major=6&micro=6.2&arch=avoton&minor=2&timezone=Amsterdam&nano=0 HTT
P/1.1" 200 50875 "-" "synology_avoton_415+ DSM6.2-23739 (package)"
```

**Comment by BenjV** [comment]
After your remarks I visited the Synology website to search for a new version of the developers guide.
There is a new guide but without a new name nor with any info about  changes or updates.
The new guide can be found here:
https://www.synology.com/en-global/support/developer#tool

I have created a pdf with  green markings of the difference with the old version.
You can find it here:
https://github.com/BenjV/SYNO-packages/raw/master/developer-guide%20dsm%206.2%20diff.pdf

Tomorrow I will studie the changes to see how it impact the repro and the current packages.

**Comment by Diaoul** [comment]
I think Synology has to make some changes with 6.2 to allow it to be backward compatible, at least to some extent.

Download count is tricky because that means handling the request first with Python then eventually redirect to the actual static file for actual download. This will increase the load on the server.

**Comment by ymartin59** [comment]
@BenjV No idea how you produced that PDF but it is terrific.
I have submitted PR from highlights: https://github.com/SynoCommunity/spksrc/pull/3363

I still do not understand magic behind the firmware registration and comparison in `spkrepo`... but I add to improvise: https://github.com/SynoCommunity/spksrc/issues/2966#issuecomment-400818796

**Comment by BenjV** [comment]
There are a lot of websites who do an online compare for you between two pdf's
It is such a big document so reading through it would certainly caused to mis something, so I automated it.

**Comment by ymartin59** [comment]
@th0ma7 May you please review this old discussion? From your point of view is there any improvement still needed in spkrepo about firmware/os version?

**Comment by th0ma7** [comment]
Good catch, indeed this should be fixed part of this PR #4219 with this commit:
* https://github.com/SynoCommunity/spksrc/commit/f5e99a3f6ade192ee7f19287d8b0037e64321c03#diff-a028235204bdbfe6f0384077e3ac6dd4556c9c79338d3a9bae68d7432acf1bf0

**Comment by th0ma7** [comment]
@ymartin59 can you please add me to the dev group?
